File: src/app/src/main/java/com/onscreensync/tvapp/MediaOnlyActivity.java
package com.onscreensync.tvapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.util.Log;
import android.widget.FrameLayout;

import com.onscreensync.tvapp.datamodels.MediaAssetDataModel;
import com.onscreensync.tvapp.fragments.ExternalMediaFragment;
import com.onscreensync.tvapp.fragments.ImageMediaFragment;
import com.onscreensync.tvapp.fragments.VideoMediaFragment;
import com.onscreensync.tvapp.signalR.SignalrHubConnectionBuilder;
import com.onscreensync.tvapp.utils.ObjectExtensions;

public class MediaOnlyActivity extends AppCompatActivity {

    FrameLayout frameLayout;
    String externalMediaSource;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_media_only);

        frameLayout = (FrameLayout) findViewById(R.id.media_only_framelayout);

        // Retrieve the Intent that started this activity
        Intent intent = getIntent();
        MediaAssetDataModel mediaAsset = intent.getParcelableExtra("mediaAsset");
        externalMediaSource = intent.getStringExtra("externalMediaSource");

        if(!ObjectExtensions.isNullOrEmpty(externalMediaSource)) {
            loadExternalMediaFragment(externalMediaSource);
        } else {
            int mediaType = mediaAsset.getType();
            switch(mediaType) {
                case 1: // Image
                    loadImageMediaFragment(mediaAsset.getAssetUrl());
                    break;
                case 2: // Video
                    loadVideoMediaFragment(mediaAsset.getAssetUrl());
                    break;
                default:
                    Log.d("MediaOnlyActivity", "No such media type");
                    break;
            }
        }
    }


    @Override
    public void onBackPressed() {
        SignalrHubConnectionBuilder.getInstance().removeConnectionFromGroup();

        final Handler handler = new Handler();
        handler.postDelayed(() -> {

            finishAffinity();
            finish();
            // Call System.exit(0) to terminate the entire process
            System.exit(0);
        }, 2000);
    }
    private void loadVideoMediaFragment(String assetUrl) {
        Bundle bundle = new Bundle();
        bundle.putString("assetUrl", assetUrl);
        Fragment fragment = new VideoMediaFragment();
        loadFragment(fragment, bundle);
    }

    private void loadImageMediaFragment(String assetUrl) {
        Bundle bundle = new Bundle();
        bundle.putString("assetUrl", assetUrl);
        Fragment fragment = new ImageMediaFragment();
        loadFragment(fragment, bundle);
    }

    private void loadExternalMediaFragment(String externalMediaSource) {
        Bundle bundle = new Bundle();
        bundle.putString("externalMediaSource", externalMediaSource);

        Fragment fragment = new ExternalMediaFragment();
        loadFragment(fragment, bundle);
    }

    private void loadFragment(Fragment fragment, Bundle bundle) {
        fragment.setArguments(bundle);

        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.media_only_framelayout, fragment);
        transaction.addToBackStack(null);
        transaction.commit();
    }
}

File: src/app/src/main/java/com/onscreensync/tvapp/MediaOnlyActivity.kt
package com.onscreensync.tvapp

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.onscreensync.tvapp.databinding.ActivityMediaOnlyBinding
import com.onscreensync.tvapp.datamodels.MediaAssetDataModel
import com.onscreensync.tvapp.fragments.ExternalMediaFragment
import com.onscreensync.tvapp.fragments.ImageMediaFragment
import com.onscreensync.tvapp.fragments.VideoMediaFragment
import com.onscreensync.tvapp.signalR.SignalRManager
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class MediaOnlyActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMediaOnlyBinding

    @Inject
    lateinit var signalRManager: SignalRManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMediaOnlyBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val mediaAsset = intent.getParcelableExtra<MediaAssetDataModel>("mediaAsset")
        val externalMediaSource = intent.getStringExtra("externalMediaSource")

        if (!externalMediaSource.isNullOrEmpty()) {
            loadExternalMediaFragment(externalMediaSource)
        } else {
            mediaAsset?.let {
                when (it.type) {
                    1 -> loadImageMediaFragment(it.assetUrl)
                    2 -> loadVideoMediaFragment(it.assetUrl)
                    else -> Log.d("MediaOnlyActivity", "No such media type")
                }
            }
        }
    }

    override fun onBackPressed() {
        signalRManager.disconnect()
        Handler(Looper.getMainLooper()).postDelayed({
            finishAffinity()
        }, 2000)
    }

    private fun loadVideoMediaFragment(assetUrl: String?) {
        val bundle = Bundle().apply {
            putString("assetUrl", assetUrl)
        }
        loadFragment(VideoMediaFragment(), bundle)
    }

    private fun loadImageMediaFragment(assetUrl: String?) {
        val bundle = Bundle().apply {
            putString("assetUrl", assetUrl)
        }
        loadFragment(ImageMediaFragment(), bundle)
    }

    private fun loadExternalMediaFragment(externalMediaSource: String) {
        val bundle = Bundle().apply {
            putString("externalMediaSource", externalMediaSource)
        }
        loadFragment(ExternalMediaFragment(), bundle)
    }

    private fun loadFragment(fragment: Fragment, bundle: Bundle) {
        fragment.arguments = bundle
        supportFragmentManager.beginTransaction()
            .replace(R.id.media_only_framelayout, fragment)
            .addToBackStack(null)
            .commit()
    }
}
